libs/langchain-openai/src/types.ts

```diff
@@ -197,4 +197,6 @@ export declare interface AzureOpenAIInput {
    * will be result in the endpoint URL: https://westeurope.api.cognitive.microsoft.com/openai/deployments/{DeploymentName}/
    */
   azureOpenAIBasePath?: string;
+
+  azureADTokenProvider?: () => Promise<string>;
```

Comment:
Could we have JSDoc here explaining what is it? I see the OpenAI SDK has some. We should at least copy that over.